File: huaweicloud-cs-scala-examples/src/main/scala/com/huaweicloud/cs/scala/model/KafkaSourceGenerator.scala
package com.huaweicloud.cs.scala.model

import org.apache.flink.streaming.api.functions.source.SourceFunction
import org.apache.flink.streaming.api.functions.source.SourceFunction.SourceContext

/**
  * This class implements an SourceFunction for Kafka to generate data source.
  */
class KafkaSourceGenerator extends SourceFunction[String] {

  var running = true
  var i = 0

  override def run(ctx: SourceContext[String]) {
    while (running) {
      ctx.collect("element-" + i)
      i += 1
      Thread.sleep(1000)
    }
  }

  override def cancel() {
    running = false
  }

}

File: huaweicloud-cs-java-examples/src/main/java/com/huaweicloud/cs/java/model/KafkaSourceGenerator.java
package com.huaweicloud.cs.java.model;

import org.apache.flink.streaming.api.functions.source.SourceFunction;

/**
 * This class implements an SourceFunction for Kafka to generate data source.
 */
public class KafkaSourceGenerator implements SourceFunction<String> {

    private static final long serialVersionUID = 2174904787118597072L;
    private boolean running = true;
    private long i = 0;

    @Override
    public void run(SourceContext<String> ctx) throws Exception {
        while (running) {
            ctx.collect("element-" + (i++));
            Thread.sleep(1000);
        }
    }

    @Override
    public void cancel() {
        running = false;
    }

}
